### ISSUE TITLE
The getter 'page' was called on null, when using newPage() after puppeteer.connect() 

### ISSUE THREAD

**Issue by javierdvalle** [xvrh/puppeteer-dart#77]
Hi, I am getting an error when using `puppeteer.connect()` instead of `puppeteer.launch()`:

The code is the following:

```dart
import 'package:puppeteer/puppeteer.dart';

main() async {
  var browser = await puppeteer.connect(browserWsEndpoint:'ws://localhost:3000');
  var page = await browser.newPage();  // error here
  await page.goto('https://example.com');
  await browser.close();
}
```

And the error:
```
Unhandled exception:
NoSuchMethodError: The getter 'page' was called on null.
Receiver: null
Tried calling: page
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:50:5)
#1      Browser._createPageInContext (package:puppeteer/src/browser.dart:181:29)
<asynchronous suspension>
#2      BrowserContext.newPage (package:puppeteer/src/browser.dart:370:20)
#3      Browser.newPage (package:puppeteer/src/browser.dart:173:28)
<asynchronous suspension>
#4      main (file:///home/javier/wkf/oferteame/pruebas/prueba-puppeteer-dart/main.dart:8:28)
<asynchronous suspension>
#5      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:301:19)
#6      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:12)
```

Package version: 
```
dependencies:
  puppeteer: ^1.11.0
```

The error raises only when using `puppeteer.connect()`, not when using `puppeteer.launch()`. 

The instance running at the websocket endpoint seems to be ok (running a javascript version of the same code works perfectly, connecting with the same websocket endpoint).

**Comment by javierdvalle** [comment]
Reading the implementation of `puppeteer.launch()` [here](https://github.com/xvrh/puppeteer-dart/blob/7cbf15343efb56a5fc94a96931a51cb608cd7f35/lib/src/puppeteer.dart#L168), I came with this workaround:

```dart
main() async {
  var browser = await puppeteer.connect(browserWsEndpoint:'ws://localhost:3000');
  //----- new code
  var targetFuture =
        browser.waitForTarget((target) => target.type == 'page');
  await browser.targetApi.setDiscoverTargets(true);
  await targetFuture;
  //--------------
  var page = await browser.newPage();
  await page.goto('https://example.com');
  await browser.close();
}
```

The problem is that after the connect(), `browser.target` is null.
Maybe puppeteer.connect() should wait for target as does puppeteer.launch()? Sorry if I am missing something, I don't understand all the details.

**Comment by xvrh** [comment]
Hello Javier,
Yes it seems there is a bug in the `connect` method. It should call `setDiscoverTargets` like `launch`.

Thank you for finding it.

I will fix it really soon and I will add more tests for the "connect" workflow.

**Comment by xvrh** [comment]
I published version 1.12.0 that should fix this issue.
